### PR TITLE
Difumo corrections

### LINKED ISSUE #2665: difumo atlas resolution
difumo maps downloaded with `resolution_mm=2` seem to have 1.875 mm resolution
```
from nilearn import datasets, image

difumo = datasets.fetch_atlas_difumo(dimension=64, resolution_mm=2)
difumo_maps = image.load_img(difumo["maps"])
print(difumo_maps.affine)
```

also why are we downloading the `resample_dictionaries.py` script?

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Closes #2665  .

Changes proposed in this pull request:

- Post merge corrections in the difumo atlas fetcher as discussed at #2665

### CODE REVIEW

File: nilearn/datasets/atlas.py [hunk 1]
```diff
@@ -32,11 +32,11 @@ def fetch_atlas_difumo(dimension=64, resolution_mm=2, data_dir=None, resume=True
     -----
     Direct download links from OSF:
 
-    - 64: https://osf.io/wjum7/download
-    - 128: https://osf.io/n3vba/download
-    - 256: https://osf.io/vza2y/download
-    - 512: https://osf.io/a23gw/download
-    - 1024: https://osf.io/jpdum/download
+    - 64: https://osf.io/pqu9r/download
+    - 128: hhttps://osf.io/wjvd5/download
```
Comment (NicolasGensollen) [inline nilearn/datasets/atlas.py:36]: Small typo here

**Review by bthirion — COMMENTED** [review]
Thx. 
Please add a whatsnew entry.

**Comment by KamalakerDadi** [comment]
> Please add a whatsnew entry.

There exists already a whats new entry.

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks!

**Review by jeromedockes — APPROVED** [review]
thanks @KamalakerDadi !

**Review by bthirion — APPROVED** [review]
Great, thx !

**Comment by bthirion** [comment]
Merging now to fix ci...

**Pull request merged**